MSAL/src/native_auth/public/MSALNativeAuthUserAccountResult+Internal.swift

```diff
@@ -26,25 +26,83 @@ import Foundation
+                                                       authority: authority)
+        config.bypassRedirectURIValidation = configuration.redirectUri == nil
+
+        guard let client = try? MSALNativeAuthPublicClientApplication(configuration: config, challengeTypes: challengeTypes)
+        else {
+            MSALLogger.log(
+                            level: .error,
+                            context: context,
+                            format: "Config or challenge types unexpectedly found nil."
+                        )
+            Task { await delegate.onAccessTokenRetrieveError(
+                error: RetrieveAccessTokenError(type: .generalError,
+                                                correlationId: correlationId ?? context.correlationId())) }
+            return
+        }
+
+        client.acquireTokenSilent(with: params) { [weak self] result, error in
+            guard let self = self else { return }
+
+            if let error = error as? NSError {
+                let accessTokenError = self.createRetrieveAccessTokenError(error: error, context: context)
+                Task { await delegate.onAccessTokenRetrieveError(error: accessTokenError) }
+                return
+            }
+
+            if let result = result {
+                let delegateDispatcher = CredentialsDelegateDispatcher(delegate: delegate, telemetryUpdate: nil)
+                let accessTokenResult = MSALNativeAuthTokenResult(accessToken: result.accessToken,
+                                                                  scopes: result.scopes,
+                                                                  expiresOn: result.expiresOn)
+                Task { await delegateDispatcher.dispatchAccessTokenRetrieveCompleted(result: accessTokenResult, correlationId: result.correlationId) }
+                return
+            }
+
+            Task { await delegate.onAccessTokenRetrieveError(error: RetrieveAccessTokenError(type: .generalError,
+                                                                                             correlationId: correlationId ?? context.correlationId())) }
         }
     }
 
-    func refreshData(authTokens: MSALNativeAuthTokens, account: MSALAccount) {
-        self.authTokens = authTokens
-        self.account = account
+    func createRetrieveAccessTokenError(error: NSError, context: MSALNativeAuthRequestContext) -> RetrieveAccessTokenError {
+        if let innerError = error.userInfo[NSUnderlyingErrorKey] as? NSError,
+           let message = innerError.userInfo[MSALErrorDescriptionKey] as? String,
+           let correlationId = correlationIdFromMSALError(error: innerError) {
+            return RetrieveAccessTokenError(type: .generalError, message: message, correlationId: correlationId, errorCodes: [])
+        }
+
+        if let message = error.userInfo[MSALErrorDescriptionKey] as? String,
+            let code = codeFromMSALError(error: error),
```

Comment:
is "MSALInternalErrorCodeKey" always going to return an internal error code? Can you please double check. If the answer is yes, we should not return this code in our error codes

Reply:
"MSALInternalErrorCodeKey" is the same for error codes, or inner error codes (although the name can be misleading). In line 90 it is going to get the error code (because it doesn't have an inner error, otherwise the function would have returned in line 86).

Reply:
If this key can also be used also for internal error, we should not return them as part of our "errorCodes" array. I'm going to remove the parse of the error code

Reply:
If "MSALInternalErrorCodeKey" can contains internal error, we should parse it. Code updated